### DISCUSSION TITLE
Code Style Help

### DISCUSSION THREAD

**Discussion by Cmeem** [prettier/prettier#14945] [Q&A] [answered]
```
const options = {
  active: true,
  ...('day' === 'hour' ? {
    hour: true
  } : {
    day: true
  })
};
```

I have the above Babel generated JS code that constructs an object property conditionally. If I prettify the code, it turns into below:

```
const options = {
  active: true,
  ...('day' === 'hour'
    ? {
        hour: true,
      }
    : {
        day: true,
      }),
};
```

Its indentation looks weird. Especially last `}),` Babel version looks more prettier, ngl.

Is this intended or a bug? Or is there any way I can improve output styling for conditional properties?

**Comment by kachkaev** [comment, marked as answer]
👋 @Cmeem!  If you remove new line after {, the result will be more compact:

 **Prettier 2.8.8**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEEAOMCW6mAvJsADpSaYCGYBAbnEpjAE4CucANJdQHSCAFAHIAJjQCeIksVIiAFhA5sZAfjKYlKluy59MAX0wtgmCZN2c4BwwEpKhgNwhuIXAXTJQNNmwgA7gAKvghoyCA0ADYBUuFuAEZsdADWcDAAyjQAtnAAMvhQcMgAZtFoPCBJqekZOHSFAObIepVw2QlwYmJdeTRQjRw0jXAAYhBs2TQwBAMRNBwwEK4gCjDZUQDqCvjwaPVgcBlhu-gMu5IRYGjxIIUVbDBByY1TpeWVAFZoAB4ZTVE4ABFDgQeDvKIVNz1NgPCIJGidKIrHBsQowTb4MQwBTIAAcAAZof4KptkjgIqi4A8mCsAI6g+DPXDhFA0NAAWiKXS6KzYcAZ+H5z2GbyQZUhlQq2XwLWsbjQALgAEEZmiEos4EE4GwCkUIVCQIqBoCQWDiuKPm4YIjMdjcUgAEzW5L4KJNADCEGyYpA1IArCsOBUACqI1kSw0MLgASSgPVgGTAaLwyvjGRgkkBBrghkMQA)

```sh
--parser babel
```

**Input:**

```jsx
const options = {
  active: true,
  ...('day' === 'hour' ? { hour: true
  } : { day: true
  })
};
```

**Output:**

```jsx
const options = {
  active: true,
  ...("day" === "hour" ? { hour: true } : { day: true }),
};

```

---

Overall, it’s best to not worry about such things and focus on code logic rather than code style.

**Reply by Cmeem** [reply]
Thank you very much for your help and great tool. I really like Prettier.

Original source code is

```
const options= {
  active: true,
  ...("day" === 'hour'
    ? { hour: true }
    : { day: true }),
};
```

It is formatted by Prettier in VS Code and I am pretty OK with it. But for production, I run Babel to remove JS comments and it seems Babel ads unnecessary new lines and therefore Prettier gets confused. 

I should live with it as you suggested. Thanks again.
